lang/cs/Org.Apache.REEF.Network/NetworkService/StreamingNetworkService.cs

```diff
@@ -42,37 +39,78 @@ public class StreamingNetworkService<T> : INetworkService<T>
         private StreamingNetworkService(
             IObserver<NsMessage<T>> messageHandler,
-            IIdentifierFactory idFactory,
             INameClient nameClient,
             StreamingRemoteManagerFactory remoteManagerFactory,
             NsMessageStreamingCodec<T> codec,
-            ILocalAddressProvider localAddressProvider,
-            IInjector injector)
+            ILocalAddressProvider localAddressProvider)
+            : this(messageHandler, null, nameClient, remoteManagerFactory, codec, localAddressProvider)
+        {
+        }
+
+        /// <summary>
+        /// Create a new Writable NetworkService
+        /// </summary>
+        /// <param name="messageHandler">The observer to handle incoming messages</param>
+        /// <param name="nameClient">The name client used to register Ids</param>
+        /// <param name="remoteManagerFactory">
+        /// Writable RemoteManagerFactory to create a Writable RemoteManager
+        /// </param>
+        /// <param name="codec">Codec for Network Service message</param>
+        /// <param name="localAddressProvider">The local address provider</param>
+        [Inject]
+        private StreamingNetworkService(
+            IObserver<IRemoteMessage<NsMessage<T>>> messageHandler,
+            INameClient nameClient,
+            StreamingRemoteManagerFactory remoteManagerFactory,
+            NsMessageStreamingCodec<T> codec,
+            ILocalAddressProvider localAddressProvider)
+            : this(null, messageHandler, nameClient, remoteManagerFactory, codec, localAddressProvider)
+        {
+        }
+
+        [Inject]
+        private StreamingNetworkService(
+            IObserver<NsMessage<T>> messageHandler1,
+            IObserver<IRemoteMessage<NsMessage<T>>> messageHandler2,
+            INameClient nameClient,
+            StreamingRemoteManagerFactory remoteManagerFactory,
+            NsMessageStreamingCodec<T> codec,
+            ILocalAddressProvider localAddressProvider)
         {
             _remoteManager = remoteManagerFactory.GetInstance(localAddressProvider.LocalAddress, codec);
 
-            // Create and register incoming message handler
-            // TODO[REEF-419] This should use the TcpPortProvider mechanism
-            var anyEndpoint = new IPEndPoint(IPAddress.Any, 0);
-            _messageHandlerDisposable = _remoteManager.RegisterObserver(anyEndpoint, messageHandler);
+            if (messageHandler1 != null)
+            {
+                // Create and register incoming message handler
+                // TODO[REEF-419] This should use the TcpPortProvider mechanism
+                var anyEndpoint = new IPEndPoint(IPAddress.Any, 0);
+                _messageHandlerDisposable1 = _remoteManager.RegisterObserver(anyEndpoint, messageHandler1);
+            }
+            else
+            {
+                _messageHandlerDisposable1 = null;
+            }
+
+            _messageHandlerDisposable2 = messageHandler2 != null ?
```

Comment:
What is messageHandler2  in REEF Network code that injected here in our scenario? Is it GroupCommNetworkObserver?